packages/utils/src/timeout.ts

```diff
@@ -1,23 +1,30 @@
-import {anySignal} from "any-signal";
-import {TimeoutError} from "./errors.js";
-import {sleep} from "./sleep.js";
+import { TimeoutError } from "./errors.js";
+import { sleep } from "./sleep.js";
 
 export async function withTimeout<T>(
   asyncFn: (timeoutAndParentSignal?: AbortSignal) => Promise<T>,
   timeoutMs: number,
   signal?: AbortSignal
 ): Promise<T> {
   const timeoutAbortController = new AbortController();
```

Comment:
One edge case left: 

In async processing, it can be a scenario if parent signal is abort before `waitTimeout` starts to execute. So we need to check that here with `signal?.aborted` before start processing. 

Also need to add one test case for this scenario as well.